### ISSUE TITLE
Use of http.RequestData makes view fail silently

### ISSUE THREAD

**Issue by mjmare** [encode/apistar#309]
I have a view, that when called will "hang". It seems that function is not entered (breakpoint not triggered, no print output, no log output, no/empty response in browser).
Removing the  "data: http.RequestData" argument will make the view function normally.

```
def upload(request: http.Request, body: http.Body, data: http.RequestData):
    result = {
        'method' : request.method,
        'url'    : request.url,
        'headers': dict(request.headers)
    }

    print('XXXX')
    log.debug(body)
    log.debug(data)
    return result
``` 
So currently there is no easy way to get the parsed body.

Using python 3.6.2, Apistar 0.3.7.

**Comment by tomchristie** [comment]
All looks resolved in latest 0.5 (substantial re-write everywhere, including `http.RequestData`, so I'll close this off now)

**Comment by GustavoRPS** [comment]
Same for me

Python 3.6.4 :: Anaconda, Inc.
apistar==0.5.32
